admin_guide/cluster_autoscaler.adoc

```diff
@@ -0,0 +1,527 @@
+
+$ aws ec2 create-image \
+    --instance-id i-012b060e213821a70 \
+    --name "${CLUSTER_ID}-ASG-golden-image-${TIMESTAMP}"
+{
+    "ImageId": "ami-01785f295539e6441"
+}
+----
+
+===== Manually creating the LC and ASG
+
+To create the LC we need the AMI of our golden image. This is either
+an existing pre-prepared "golden" image or the AMI
+(`ami-01785f295539e6441`) we created in the previous section.
+
+[source,bash]
+----
+$ aws autoscaling create-launch-configuration
+      --launch-configuration-name ${CLUSTER_ID}-LC \
+      --image-id ami-01785f295539e6441 \
+      --instance-type m4.large \
+      --key-name libra
+----
+
+[source,bash]
+----
+$ aws autoscaling create-auto-scaling-group \
+      --auto-scaling-group-name ${CLUSTER_ID}-ASG \
+      --launch-configuration-name ${CLUSTER_ID}-LC \
+      --min-size 0 \
+      --max-size 6 \
+      --vpc-zone-identifier subnet-cf57c596 \
+      --tags ResourceId=${CLUSTER_ID}-ASG,ResourceType=auto-scaling-group,Key=Name,Value=${CLUSTER_ID}-ASG-node,PropagateAtLaunch=true ResourceId=${CLUSTER_ID}-ASG,ResourceType=auto-scaling-group,Key=kubernetes.io/cluster/${CLUSTER_ID},Value=true,PropagateAtLaunch=true
+----
+
+==== Deploying the auto-scaler components
+
+Once the cloud-provider specific components have been created (i.e.,
+the ASG and LC) we can run the ansible playbook to automatically
+deploy the remaining cluster auto-scaler components (e.g., Service
+Account, RBAC roles, the cluster-autoscaler deployment itself, etc)
+but we need to specify some additional configuration:
+
+.OpenShift extra variables for the openshift-cluster-autoscaler playbook
+----
+# We need to specify the ASG by name so the deployment can reference it
+openshift_cluster_autoscaler_node_groups:
+- min: 0
+  max: 6
+  name: mycluster-ASG
+
+# We want new nodes to be automatically approved
+openshift_master_bootstrap_auto_approve: true
+
+# We want to install the cluster-autoscaler components
+openshift_cluster_autoscaler_install: True
+----
+
+Now run the openshift-cluster-autoscaler playbook:
+
+----
+# ansible-playbook -i <inventory_file> \
+ifdef::openshift-enterprise[]
+    /usr/share/openshift-ansible/playbooks/openshift-cluster-autoscaler/config.yml \
+endif::[]
+ifdef::openshift-origin[]
+    ~/openshift-ansible/playbooks/openshift-cluster-autoscaler/config.yml \
+endif::[]
+    -e @asg-vars.yaml
+----
+
+When this completes verify that there is a cluster-autoscaler deployment:
+
+[source,bash]
+----
+$ oc get deployment
+NAME                 DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+cluster-autoscaler   1         1         1            1           23h
+----
+
+==== Cleaning up the ASG and LC
+
+When you no longer need auto-scaling capability you can delete the
+deployment on the cluster:
+
+[source,bash]
+----
+$ oc delete deployment cluster-autoscaler -n openshift-autoscaler
+----
+
+And the AWS resources can be deleted as long as there are no
+outstanding nodes provisioned through the ASG:
```

Comment:
How do you tear down the nodes that you provisioned through the ASG? (That needs to be a separate step before this one.)

Reply:
I think so.